validations/src/validations.rs

```diff
@@ -645,12 +626,34 @@ where
     Hash::from(crypto_merkle_tree_root(&transactions_hashes))
 }
 
+/// Function to calculate a merkle tree from a transaction vector
+pub fn hash_merkle_tree_root(hashes: &[Hash]) -> Hash {
+    let hashes: Vec<Sha256> = hashes
+        .iter()
+        .map(|x| match x {
+            Hash::SHA256(x) => Sha256(*x),
+        })
+        .collect();
+
+    Hash::from(crypto_merkle_tree_root(&hashes))
+}
+
 /// Function to validate block's merkle tree
 pub fn validate_merkle_tree(block: &Block) -> bool {
-    let merkle_tree = block.block_header.hash_merkle_root;
-    let transactions = &block.txns;
-
-    merkle_tree == merkle_tree_root(transactions)
+    // Compute `hash_merkle_root` and build block header
+    let mint_hash_merkle_root = merkle_tree_root(&[block.txns.mint.clone()]);
+    let vt_hash_merkle_root = merkle_tree_root(&block.txns.value_transfer_txns);
+    let dr_hash_merkle_root = merkle_tree_root(&block.txns.data_request_txns);
+    let commit_hash_merkle_root = merkle_tree_root(&block.txns.commit_txns);
+    let reveal_hash_merkle_root = merkle_tree_root(&block.txns.reveal_txns);
+    let tally_hash_merkle_root = merkle_tree_root(&block.txns.tally_txns);
+
+    (mint_hash_merkle_root == block.block_header.merkle_roots.mint_hash_merkle_root)
```

Comment:
You could create a new `BlockMerkleRoots` and just compare it with `block.block_header.merkle_roots`